### ISSUE TITLE
Raw query

### ISSUE THREAD

**Issue by fabreax** [pardom-zz/ActiveAndroid#181]
Hello, I need to do a "SELECT DISTINCT column FROM MyTable"

I tried ActiveAndroid.getDatabase().rawQuery("SELECT DISTINCT column FROM MyTable", null);
but I have a "java.lang.NoSuchMethodError: android.database.sqlite.SQLiteDatabase.query"

Is there a solution ?

Thank you.

**Comment by fabreax** [comment]
Hello,

Thank you but "execSQL" has no value returned.

2014-02-03 Leonardo Rossetto notifications@github.com:

**Comment by omaraf** [comment]
Did you ever find a solution to this issue?

**Comment by joshuapinter** [comment]
Have you tried using the `distinct()` method? e.g `new Select().distinct()`

**Comment by omaraf** [comment]
Yes, but where should I set which column to filter distincts? For example, to count how many distinct values are on a specific column?—

**Comment by fabreax** [comment]
Hello, I have not found any solution, I have switched to another ORM framework.

**Comment by Maragues** [comment]
Did you try SQLiteUtils.rawQuery ? https://github.com/pardom/ActiveAndroid/blob/master/src/com/activeandroid/util/SQLiteUtils.java

``` java
public static <T extends Model> List<T> rawQuery(Class<? extends Model> type, String sql, String[] selectionArgs) {
        Cursor cursor = Cache.openDatabase().rawQuery(sql, selectionArgs);
        List<T> entities = processCursor(type, cursor);
        cursor.close();
        return entities;
    }
```

**Comment by melwinm** [comment]
Von meinem Sony Xperia™-Smartphone gesendet

---- Miguel Aragues schrieb ----

**Comment by roozevasl** [comment]
You should use Distinct as below:

List<Data> datas = new Select()
                .distinct()
                .from(Data.class)
                .groupBy("course_type")
                .execute();
